File: backend/PropMate.Api/Models/Property.cs
namespace PropMate.Api.Models;

public class Property
{
    public int Id { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public decimal Price { get; set; }
    public int Bedrooms { get; set; }
    public int Bathrooms { get; set; }
    public string Location { get; set; } = string.Empty;
    public double Latitude { get; set; }
    public double Longitude { get; set; }
    public bool IsAvailable { get; set; }
}

File: mobile/propmate_mobile/lib/models/property.dart
class Property {
  final int id;
  final String title;
  final String description;
  final double price;
  final int bedrooms;
  final int bathrooms;
  final String location;
  final double latitude;
  final double longitude;
  final bool isAvailable;

  Property({
    required this.id,
    required this.title,
    required this.description,
    required this.price,
    required this.bedrooms,
    required this.bathrooms,
    required this.location,
    required this.latitude,
    required this.longitude,
    required this.isAvailable,
  });

  factory Property.fromJson(Map<String, dynamic> json) {
    return Property(
      id: json['id'],
      title: json['title'],
      description: json['description'],
      price: (json['price'] as num).toDouble(),
      bedrooms: json['bedrooms'],
      bathrooms: json['bathrooms'],
      location: json['location'],
      latitude: (json['latitude'] as num).toDouble(),
      longitude: (json['longitude'] as num).toDouble(),
      isAvailable: json['isAvailable'],
    );
  }
}
